src/main/java/org/openRealmOfStars/player/tech/TechList.java

```diff
@@ -902,50 +825,61 @@ public double getTechResearchPoints(final TechType type) {
   public void setTechResearchPoints(final TechType type, final double value) {
     techResearchPoint[type.getIndex()] = value;
   }
+
   /**
    * Add new random tech to player tech list
    * @param info PlayerInfo
    * @return Added Tech or null if all tech has been invented
    */
   public Tech addNewRandomTech(final PlayerInfo info) {
     Tech result = null;
-    int loop = 10;
-    while (result == null && loop > 0) {
-      loop--;
+    for (int j = 0; result == null && j < 10; j += 1) {
```

Comment:
`j < 10` probably could be` j < MAX_TECH_LEVELS` also `j += 1` could be `j++`. It is shorter than easier to understand. For me that` j += 1` causes always, wait what is this :smile: